src/rars/tools/ConversionTool.java

```diff
@@ -0,0 +1,242 @@
+                updateFromDecimal(decimalField, hexField, binaryField, charField);
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                updateFromDecimal(decimalField, hexField, binaryField, charField);
+            }
+        });
+
+        hexField.getDocument().addDocumentListener(new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                updateFromHex(hexField, decimalField, binaryField, charField);
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                updateFromHex(hexField, decimalField, binaryField, charField);
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                updateFromHex(hexField, decimalField, binaryField, charField);
+            }
+        });
+
+        binaryField.getDocument().addDocumentListener(new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                updateFromBinary(binaryField, decimalField, hexField, charField);
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                updateFromBinary(binaryField, decimalField, hexField, charField);
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                updateFromBinary(binaryField, decimalField, hexField, charField);
+            }
+        });
+
+        charField.getDocument().addDocumentListener(new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                updateFromChar(charField, decimalField, hexField, binaryField);
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                updateFromChar(charField, decimalField, hexField, binaryField);
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                updateFromChar(charField, decimalField, hexField, binaryField);
+            }
+        });
+    }
+
+    /**
+     * Updates the other fields when the decimal field is changed.
+     * 
+     * @param decimalField      The decimal text field.
+     * @param hexField          The hexadecimal text field.
+     * @param binaryField       The binary text field.
+     * @param charField         The character text field.
+     */
+    private void updateFromDecimal(JTextField decimalField, JTextField hexField, JTextField binaryField, JTextField charField) {
+        if (updating) return;
+        updating = true;
+        try {
+            int decimal = Integer.parseInt(decimalField.getText());
+            hexField.setText("0x" + Integer.toHexString(decimal));
+            binaryField.setText("0b" + Integer.toBinaryString(decimal));
+            charField.setText(Character.toString((char) decimal));
```

Comment:
Maybe just write "" if the number > 127 ; above 127 it's not strictly ASCII. Moreover, by casting to char this wrap for number > 65535 (so it's not strictly Unicode either)